test/test_check_branch.py

```diff
@@ -70,3 +70,16 @@ def expect_exception(branch, deployment, message):
                              'prod',
                              "Branch 'feature/foo' cannot be deployed to 'prod', "
                              "only one of {'sandbox'} or personal deployments.")
+
+        branch_context = script.branch_context
+        supported_vars = ['GITHUB_BASE_REF', 'CI_COMMIT_REF_NAME', 'GITHUB_HEAD_REF']
+        with patch.object(os, 'environ', os.environ | dict.fromkeys(supported_vars)):
```

Comment:
This line seems off to me. First, why not use `patch.dict` like on line 84? Second, the test still passes with this line completely removed, which I think it should be. I don't even understand what it would mean for an environment variable to be set to `None`. If a variable isn't set, I'd expect accessing it to raise a `KeyError`. That's why the script uses `.get`.

Reply:
If the intent is to cover cases where multiple variables are set at once, you'd need to use a non-null value. Empty string might make sense.

Reply:
I've added a comment to the patch, hopefully it's a bit more straight forward.